lib/cib/cib_ops.c

```diff
@@ -453,14 +453,16 @@ update_cib_object(xmlNode * parent, xmlNode * update)
 
     copy_in_properties(target, update);
 
-    crm_trace("Processing children of <%s id='%s'>",
-              pcmk__s(object_name, "<null>"), pcmk__s(object_id, "<null>"));
+    crm_trace("Processing children of <%s%s%s>", object_name,
+              ((object_id == NULL)? "" : " id="), pcmk__s(object_id, ""));
 
     for (a_child = pcmk__xml_first_child(update); a_child != NULL;
          a_child = pcmk__xml_next(a_child)) {
         int tmp_result = 0;
 
-        crm_trace("Updating child <%s id=%s>", crm_element_name(a_child), ID(a_child));
+        crm_trace("Updating child <%s%s%s>", crm_element_name(a_child),
+                  ((ID(a_child) == NULL)? "" : " id="),
+                  pcmk__s(ID(a_child), ""));
 
```

Comment:
shouldn't we be careful with multiple usage of ID here as it logs lots of errors?

Reply:
I'm not worried since it's only evaluated if tracing is enabled for this function, which would only be while debugging